### ISSUE TITLE
Make using stable or nightly configurable

### ISSUE THREAD

**Issue by danreeves** [yoshuawuyts/crossgen#22]
# Feature Request
## Summary
Add a flag which toggles between generating stable or nightly config.

## Motivation
Currently crossgen only supports generating config to test on nightly. If someone want's to test on stable instead they have to manually change all the places which say nightly.

## Expected Behavior
`crossgen`: current default is to generate nightly config
`crossgen --stable`: create config for stable

There's currently only a handful of places where a switch would be needed. In `.travis.yml` and in https://github.com/yoshuawuyts/crossgen/blob/master/templates/lib/script.sh#L22

## Drawbacks
Not sure how it might be implemented: some sort of templating or just bash conditionals? 

## Rationale and alternatives
The only other alternative is manual changes after running crossgen

## Unresolved Questions
Should stable or nightly be the default. I'd consider changing this a breaking change.

**Comment by yoshuawuyts** [comment]
I'd argue we could probably default to stable, and perhaps add an extra check for `beta`? `nightly` on CI has traditionally seemed rather volatile, and was only a requirement because until recently `Clippy` wouldn't work on any other channel. But now that it does I reckon we can probably switch?

**Comment by danreeves** [comment]
Sounds good to me! If we include `beta` does it make sense to just use a `--channel` flag which defaults to stable?

```sh
crossgen --channel beta
crossgen --channel nightly
# vs
crossgen --beta
crossgen --nightly
```

**Comment by yoshuawuyts** [comment]
@danreeves yeah, I think so! It'd also be cool if we could write them as shorthands too:
```sh
$ crossgen -c stable
```

---

@danreeves it's probably also worth considering how much of channels we're testing. E.g. for a default configuration we'd probably want something like:

- test on `{linux,windows,macOS,BSD}` on stable
- test `linux` on beta

There's probably a balance to be found in how many tests we run.

**Comment by danreeves** [comment]
pls no @stale bot 🙏
